### PR TITLE
Feat: add a new 'updatecli()' shared library

### PR DESCRIPTION
**Pull request by dduportal**
This PR adds a new shared library to allow executing `updatecli` in your pipelines.

Example:

```groovy
@Library('pipeline-library@pull/200/head') _

parallel(
  failFast: true,
  'docker-image': {
    // Do something...
  },
  'updatecli': {
    withCredentials([string(credentialsId: 'updatecli-github-token', variable: 'UPDATECLI_GITHUB_TOKEN')]) {
      updatecli(action: 'diff')
      if (env.BRANCH_NAME == 'main') {
        updatecli(action: 'apply')
      }
    }
  },
)
```

### CODE REVIEW

File: vars/updatecli.groovy [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def call(userConfig = [:]) {
+  def defaultConfig = [
+    action: 'diff',
+    config: './updatecli/updatecli.d',
+    values: './updatecli/values.yaml',
+    updatecliDockerImage: 'ghcr.io/updatecli/updatecli:latest',
+  ]
+
+  // No Deep Merge - https://e.printstacktrace.blog/how-to-merge-two-maps-in-groovy/
+  final Map finalConfig = defaultConfig + userConfig
```
Comment (timja) [inline vars/updatecli.groovy:11]: not sure if any advantage, but we normally use this shorter form: https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/notifyBuildEvent.groovy#L19-L22

Comment (dduportal) [reply vars/updatecli.groovy:11]: Thanks @timja ! I have to admit that, even though I understand with your comment that this code is somehow merging the 2 maps, I'm not sure to really understand the operator `<<` . The Groovy documentation at https://groovy-lang.org/operators.html#_bit_shift_operators tells me that it's a shift operator, but how does it work with a Map ? Is there an override in the Map class or something?

**Review by olblak — DISMISSED** [review]
I am quite excited to easily it \o/ from our infra

File: vars/updatecli.groovy (hunk 1 above)
Comment (timja) [reply vars/updatecli.groovy:11]: it's called `leftShift()` https://blog.mrhaki.com/2010/04/groovy-goodness-adding-maps-to-map_21.html

Comment (dduportal) [reply vars/updatecli.groovy]: @timja Switched to your proposal, with a link to the page you mentioned. Thanks!

**Review by timja — APPROVED** [review]

**Pull request merged**